### PR TITLE
Fix case-insensitive listing.

### LINKED ISSUE #5: Six tests are failing on Windows
The test test/test_list.dart has 6 subtests failing on Windows.  The failures can be seen at
 https://build.chromium.org/p/client.dart.packages/builders/packages-windows-glob/

### PR DESCRIPTION
**Pull request by nex3**
This also marks a test as skipped to get all tests passing.

Closes #5

### CODE REVIEW

File: lib/src/list_tree.dart [hunk 1]
```diff
@@ -332,35 +330,57 @@ class _ListTreeNode {
...
+                error.osError.errorCode == _ENOENT_WIN);
+          });
+          resultGroup.add(stream);
+        });
+      }
+      resultController.close();
+      resultGroup.close();
+      return resultGroup.stream;
+    }());
   }
```
Comment (natebosch) [inline lib/src/list_tree.dart]: [optional] Method is going from ~50 lines to ~70 lines. Could be easier to read if some chunks of it were pulled out into separate methods.

File: lib/src/list_tree.dart [hunk 2]
```diff
@@ -332,35 +330,57 @@ class _ListTreeNode {
...
-        });
-        resultGroup.add(stream);
-      });
-    },
-        onError: resultController.addError,
-        onDone: () {
-          resultController.close();
-          resultGroup.close();
-        });
+          /// If there are no [entities] that match [sequence], manually list
```
Comment (natebosch) [inline lib/src/list_tree.dart]: [nit] this is a block comment not a doc comment. How about //

File: lib/src/list_tree.dart [hunk 3]
```diff
@@ -332,35 +330,57 @@ class _ListTreeNode {
...
+          /// the directory to force `dart:io` to throw an error. This allows us
+          /// to ensure that listing "foo/bar/*" fails on case-sensitive systems
+          /// if "foo/bar" doesn't exist.
+          await child.list(
+              p.join(dir, (sequence.nodes.single as LiteralNode).text),
+              followLinks: followLinks).toList();
```
Comment (natebosch) [inline lib/src/list_tree.dart]: [optional] It's a micro-optimization, but would `last` use less memory than `toList()`?

File: lib/src/list_tree.dart [hunk 4]
```diff
@@ -377,16 +397,35 @@ class _ListTreeNode {
...
+
+    if (!_caseSensitive) {
+      children.forEach((sequence, child) {
+        if (!child._isIntermediate) return;
+        if (entities.any((entity) =>
+            sequence.matches(p.relative(entity.path, from: dir)))) {
+          return;
+        }
+
+        /// If there are no [entities] that match [sequence], manually list the
```
Comment (natebosch) [inline lib/src/list_tree.dart]: [nit] double slash?

File: lib/src/list_tree.dart (hunk 2 above)
Comment (nex3) [inline lib/src/list_tree.dart]: Done.

File: lib/src/list_tree.dart (hunk 1 above)
Comment (nex3) [inline lib/src/list_tree.dart]: Done.

File: lib/src/list_tree.dart (hunk 4 above)
Comment (nex3) [inline lib/src/list_tree.dart]: Done.

**Pull request merged**